script/install.sh

```diff
@@ -34,15 +34,47 @@ fatal() {
     exit 1
 }
 
-curl() {
+# ------------------------------------------------------------------------------
+curl2() {
     : ${CURL_USER_AGENT:="installer/install.sh $(command curl --version | head -n 1)"}
+    TIMEOUT_ARGS="--connect-timeout 5 --speed-time 30 --speed-limit 1024"
+    command curl $TIMEOUT_ARGS -fL -A "$CURL_USER_AGENT" "$@"
+}
+
+curl() {
     if [ "$verbosity" -gt 0 ]; then
-        command curl -f#L --retry 3 -A "$CURL_USER_AGENT" "$@"
+        curl2 -# "$@"
     else
-        command curl -fsSL --retry 3 -A "$CURL_USER_AGENT" "$@"
+        curl2 -sS "$@"
     fi
 }
 
+# url path
+download() {
+    local url=$1
+    local path=$2
+
+    for i in {0..4}; do
+        if curl "$url" -o "$path"; then
+            break
+        elif [ $i -lt 4 ]; then
+            sleep $((1 << $i))
+        else
+            fatal "Failed to download '$url'"
```

Comment:
Should we try to cleanup intermediate temp files on failure?
(due to `fatal` exiting directly the cleanup in `fetch` won't take place)